### PR TITLE
The solution of the 'unique cars' problem

### PR DESCRIPTION
**Pull request by koryun23**

### CODE REVIEW

File: src/main/com/aca/homework/week11/unique/cars/Car.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public int hashCode() {
+        return Objects.hash(brand, model);
+    }
+
+
+    @Override
+    public int compareTo(Car o) {
+        int result = this.brand.compareTo(o.getBrand());
+        if(result == 0){
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/Car.java:45]: please format the code

File: src/main/com/aca/homework/week11/unique/cars/UniqueCars.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+public class UniqueCars {
+
+    private final Supplier<String> supplier;
+
+    public UniqueCars(Supplier<String> supplier) {
+        this.supplier = supplier;
+    }
+
+    public Set<Car> askAndCalculate() {
+        Set<Car> cars = new HashSet<>();
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/UniqueCars.java:18]: to have sorted set of cars you should use tree set

File: src/main/com/aca/homework/week11/unique/cars/Result.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.aca.homework.week11.unique.cars;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+public class Result {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/Result.java:8]: not sure that we need this class if you use sorted set instead of hash set

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
modified

File: src/main/com/aca/homework/week11/unique/cars/UniqueCars.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public UniqueCars(Supplier<String> supplier) {
+        this.supplier = supplier;
+    }
+
+    public Set<Car> askAndCalculate() {
+        Set<Car> cars = new TreeSet<>(new Comparator<Car>() {
+            @Override
+            public int compare(Car o1, Car o2) {
+                int result = o1.getBrand().compareTo(o2.getBrand());
+                if(result == 0){
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/UniqueCars.java:19]: please format the code

File: src/main/com/aca/homework/week11/unique/cars/UniqueCars.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+public class UniqueCars {
+
+    private final Supplier<String> supplier;
+
+    public UniqueCars(Supplier<String> supplier) {
+        this.supplier = supplier;
+    }
+
+    public Set<Car> askAndCalculate() {
+        Set<Car> cars = new TreeSet<>(new Comparator<Car>() {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/UniqueCars.java:15]: let's create a nested private static comparator implementation class

File: src/main/com/aca/homework/week11/unique/cars/UniqueCars.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            );
+        }
+        return cars;
+    }
+
+    public void printAllData(Set<Car> cars) {
+        System.out.println("Number of unique brand and models: " + numberOfUniqueCars(cars) + "\n" + allCarsToString(cars));
+    }
+
+    private static String allCarsToString(Set<Car> cars) {
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week11/unique/cars/UniqueCars.java:48]: here you load your set in a string object which is not efficient. instead you could just print each element in a loop

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

**Pull request merged**